### ISSUE TITLE
CSRF protection is too restrictive

### ISSUE THREAD

**Issue by arnaud-morvan** [camptocamp/c2cgeoportal#2904]
Actually the CSRF protection is : 
```
if not (
    request.path_info_peek() == "admin" and request.referer is None or
    is_valid_referer(request.referer, settings)
):
```
But this is too much restrictive as this apply to all pyramid protected views, including GET requests.

The ``admin`` view in not the only one that need to be accessible without referer.
For example for aeroportdelyon_sig, we have protected custom apps (need authentication), and we also need authentication for the standard desktop and mobile app.

For me this should be simply : 
```
if not (request.referer is None or is_valid_referer(request.referer, settings)):
```
We could be more restrictive for non GET requests and AJAX request.

**Comment by arnaud-morvan** [comment]
@pvalsecc @sbrunner : How do you feel about this ?

**Comment by sbrunner** [comment]
I think that we should add a white list of path where we should be less restrictive...

**Comment by arnaud-morvan** [comment]
another option could be to add a view decorator where CSRF protection is needed.

**Comment by sbrunner** [comment]
By default it should be everywhere, and is should be disable where and when it's needed...

**Comment by arnaud-morvan** [comment]
Patrick have check the valid ones (forgot None) and you have added (admin routes and no referer). IMHO it is not necessary to have a white list, checking that referer is None or valid one should be enough.

**Comment by arnaud-morvan** [comment]
@sbrunner : This have been invalidated by #2915
We cannot anymore use links pointing to GMF.
Can you explain the difference between request.referer is None and request.referer = "".

**Comment by arnaud-morvan** [comment]
Note that in aeroportdelyon_sig, user is autenticated by a kerberos token, and viewer (desktop/mobile) is not accessible to unauthenticated users. So this result in an 403 for all links pointing to c2cgeoportal.

**Comment by sbrunner** [comment]
For me it's fixed on 2.3
